src/soc/soc.c

```diff
@@ -23,6 +23,8 @@
 static struct soc_t *socs = NULL;
 
 void soc_register(struct soc_t **soc, char *brand, char *type) {
+	int i;
```

Comment:
I will change the variable definition with initial value. 
Do you have any more comments?